http-server/src/main/java/org/triplea/server/error/reporting/CreateIssueStrategy.java

```diff
@@ -9,27 +11,56 @@
 import org.triplea.http.client.github.issues.GithubIssueClient;
 import org.triplea.http.client.github.issues.create.CreateIssueResponse;
 
+import com.google.common.annotations.VisibleForTesting;
+
 import lombok.Builder;
+import lombok.NonNull;
 
 /** Performs the steps for uploading an error report from the point of view of the server. */
 @Builder
 public class CreateIssueStrategy implements Function<ErrorReportRequest, ErrorUploadResponse> {
 
+  @VisibleForTesting
+  static final String STUBBED_RETURN_VALUE = "API-token==test--returned-a-stubbed-github-issue-link";
+
   @Nonnull
   private final Function<CreateIssueResponse, ErrorUploadResponse> responseAdapter;
   @Nonnull
   private final GithubIssueClient githubIssueClient;
   @Nonnull
-  private final Predicate<ErrorReportRequest> allowErrorReport;
+  private final Predicate<String> allowErrorReport;
+  /**
+   * The 'production' flag is to help us verify we are not in a 'test' mode and will not return
+   * stubbed values while in production.
+   */
+  @NonNull
+  private final Boolean isProduction;
+  @Nonnull
+  private final ErrorReportingDao errorReportingDao;
+
 
   @Override
   public ErrorUploadResponse apply(final ErrorReportRequest errorReportRequest) {
-    if (allowErrorReport.test(errorReportRequest)) {
-      final CreateIssueResponse response =
-          githubIssueClient.newIssue(errorReportRequest.getErrorReport());
-      return responseAdapter.apply(response);
+    if (allowErrorReport.test(errorReportRequest.getClientIp())) {
```

Comment:
This code is completing the error upload feature, it's the server side throttle that limits how many error uploads can be submitted (per IP address).